components/feature/downloads/src/main/java/mozilla/components/feature/downloads/share/ShareDownloadMiddleware.kt

```diff
@@ -0,0 +1,180 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.downloads.share
+
+import android.content.Context
+import android.webkit.MimeTypeMap
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.WorkerThread
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeout
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.DownloadAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.content.DownloadState
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Request
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.android.content.shareMedia
+import mozilla.components.support.ktx.kotlin.sanitizeURL
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import kotlin.math.absoluteValue
+import kotlin.random.Random
+
+/**
+ * 200 response code for a successful network request.
+ */
+@VisibleForTesting
+internal const val OK_STATUS = 200
```

Comment:
Nit: We could use `Response.SUCCESS` https://github.com/mozilla-mobile/android-components/blob/c1a85a4254a7a79eab60fe3c71a67e15f7db101f/components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Response.kt#L131 :)

Reply:
Thanks!